simulator/risc_v/riscv_decoder.h

```diff
@@ -63,40 +110,115 @@ struct RISCVInstrDecoder
         switch (subset) {
         case 'I': return I_imm;
         case 'B': return get_B_immediate();
-        case 'S': return S_imm4_0 | (S_imm11_5 << 5);
+        case 'S': return S_imm4_0 | (S_imm11_5 << 5U);
         case 'U': return U_imm;
         case 'J': return get_J_immediate();
         case 'C': return csr_imm;
         case ' ': return 0;
-        default: assert(0); return 0;
+        default: return get_compressed_immediate_value( subset);
+        }
+    }
+
+    uint32 get_compressed_immediate_value( char subset) const noexcept
+    {
+        switch (subset) {
```

Comment:
In previous review I asked to think about readability of that function.

Reply:
I've renamed immediates relatively to instruction formats containing them. Do I need to change something else? I have so many cases in switch/case, because, for example, LDSP and LWSP have different methods to obtain the immediate, while these both instructions are of CI-format.

Reply:
Let's start with extraction of the each case to a separate function.
We have to be a little more pedantic regarding code quality here, as it is a fundamental piece of code. If it is written in a clear way, it will be easy to expand it later with new RISC-V instruction subsets.